nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/TwitterStreamAPI.java

```diff
@@ -0,0 +1,174 @@
+        assert context != null;
+        assert queue != null;
+        assert logger != null;
+
+        this.queue = queue;
+        this.logger = logger;
+
+        this.tweetFields = parseCommaSeparatedListPropreties(context, TWEET_FIELDS_PROPERTY_NAME);
+        this.userFields = parseCommaSeparatedListPropreties(context, USER_FIELDS_PROPERTY_NAME);
+        this.mediaFields = parseCommaSeparatedListPropreties(context, MEDIA_FIELDS_PROPERTY_NAME);
+        this.pollFields = parseCommaSeparatedListPropreties(context, POLL_FIELDS_PROPERTY_NAME);
+        this.placeFields = parseCommaSeparatedListPropreties(context, PLACE_FIELDS_PROPERTY_NAME);
+        this.expansions = parseCommaSeparatedListPropreties(context, EXPANSIONS_PROPERTY_NAME);
+        this.backfillMinutes = context.getProperty(BACKFILL_MINUTES_PROPERTY_NAME).asInteger();
+
+        TwitterCredentialsBearer creds = new TwitterCredentialsBearer(context.getProperty(BEARER_TOKEN_PROPERTY_NAME).getValue());
+        api = new TwitterApi();
+        api.setTwitterCredentials(creds);
+
+        this.executorService = Executors.newSingleThreadExecutor();
+    }
+
+    public String getBasePath() {
+        return api.getApiClient().getBasePath();
+    }
+
+    public void setBasePath(final String path) {
+        api.getApiClient().setBasePath(path);
+    }
+
+    /**
+     * This method would be called when we would like the stream to get started. This method will spin off a thread that
+     * will continue to queue tweets on to the given queue passed in the constructor. The thread will continue
+     * to run until {@code stop} is called.
+     * @param endpoint {@code TwitterStreamAPI.SAMPLE_ENDPOINT} or {@code TwitterStreamAPI.SEARCH_ENDPOINT}
+     */
+    public void start(final String endpoint) {
+        try {
+            if (endpoint.equals(SAMPLE_ENDPOINT)) {
+                stream = api.tweets().sampleStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else if (endpoint.equals(SEARCH_ENDPOINT)) {
+                stream = api.tweets().searchStream(expansions, tweetFields, userFields, mediaFields, placeFields, pollFields, backfillMinutes);
+            } else {
+                throw new ProcessException("Endpoint was invalid value: " + endpoint);
+            }
+        } catch (ApiException e) {
+            logger.error("Received error {}: {}", e.getCode(), e.getMessage());
+        }
+
+        executorService.execute(new TweetQueuer());
+    }
+
+    /**
+     * This method would be called when we would like the stream to get stopped. The stream will be closed and the
+     * executorService will be shut down. If it fails to shutdown, then it will be forcefully terminated.
+     */
+    public void stop() {
+        try {
+            stream.close();
+        } catch (IOException e) {
+            logger.error("IOException occurred while closing stream: {}", e.getMessage());
```

Comment:
Instead of naming the exception itself, recommend simplifying the message and include the exception to provide the full stack trace:
```suggestion
            logger.error("Closing response stream failed", e);
```

Reply:
Making the change